frontend-react/src/pages/admin/value-set-editor/ValueSetsDetail.tsx

```diff
@@ -34,37 +36,23 @@ const valueSetColumns: ColumnConfig[] = [
     },
 ];
 
-const defaultValueSetRows = [
-    {
-        display: "American Indian or Alaska Native",
-        code: "1002-5",
-        version: "2.5.1",
-        system: "HL7",
-    },
-    {
-        display: "Asian",
-        code: "2028-9",
-        version: "2.5.4",
-        system: "Name of org",
-    },
-    {
-        display: "Black or African American",
-        code: "2054-5",
-        version: "2.3.0",
-        system: "HL7",
-    },
-];
-
-/* END OF FAUX DATA AND STUFF TO BE REMOVED WHEN IMPLEMENTING THE API */
-
+const createUseValueSetsRowTable = (valueSetName: string) => {
+    return generateUseLookupTable<ValueSetRow>(
+        LookupTables.VALUE_SET_ROW,
+        valueSetName
+    );
+};
 const ValueSetsDetailTable = ({ valueSetName }: { valueSetName: string }) => {
-    /* This would be replaced by our API response as reactive state (useResource) */
+    const useValueSetsRowTable = useMemo(
+        () => createUseValueSetsRowTable(valueSetName),
+        [valueSetName]
```

Comment:
@sethdarragile6 Just talked to @ahay-agile6 about it and it turns out I think the solution is calling the hook here and then just eslint-ignore the dependency array. :) Should clean up the extra abstractions and only call once.

Reply:
Also CC: @doug-s-nava! My bad

Reply:
so, maybe we're saying the same thing here, but looking at this I think maybe we are making things more complicated than they need to be. Is this what you're suggesting, @kevinhaube?

```suggestion
    const useValueSetsRowTable = useMemo(
        () =>
            generateUseLookupTable<ValueSetRow>(
                LookupTables.VALUE_SET_ROW,
                valueSetName
            ),
        [valueSetName]
    );

```
The reason `generateUseLookupTable` was originally used on the index page was as a way to eliminate a constant value from the equation (the `tableName` parameter), but in the detail page case we're passing in a more dynamic `dataSetName` as well, which can only be known once the component renders. In that context, the `createUseValueSetsRowTable` is there to theoretically extend the partial application idea by still closing over the `tableName` so that we don't have to reference the constant in the render. There are better ways to do that though, and we don't gain much from it, so I think going with this is a good idea to simplify things

Reply:
Ahh, I think I understand! In that case, keep the utility of `createUseValueSetsRowTable` but rename it to just `useValueSetsRowTable` to abide by hook name conventions in React, and what's now known as `generateUseLookupTable` would be the only redundant piece. The `useMemo` would go in `useValueSetsRowTable`.

```typescript
const createUseValueSetsRowTable = (valueSetName: string) => {
    return generateUseLookupTable<ValueSetRow>(
        LookupTables.VALUE_SET_ROW,
        valueSetName
    );
};

// ^ that becomes this...

const useValueSetsRowTable = (valueSetName: string) => 
    useMemo(() => useLookupTable<T>(
        LookupTables.VALUE_SET_ROW, 
        valueSetName
    ),
    [] // eslint-ignore if it warns about dependencies. Take a stand against the AI overlords. 
)
// Only renders once because useMemo wraps it as a stable reference and no dependencies can refresh it!
// Simply consume:
const table = useValueSetsRowTable("tableName")
```

Reply:
@kevinhaube maybe you me and @sethdarragile6 can get on a call later to hash this out, I'm getting a little bit confused about how your proposal would work